### DISCUSSION TITLE
(Proposal) Start OpenToonz with all panels locked

### DISCUSSION THREAD

**Discussion by DarrenTAnims** [opentoonz/opentoonz#6367] [Ideas]
As one of many people that try to help newcomers to OpenToonz, I see regularly a problem with someone has tried resizing a panel or just clicked and dragged near the edge of a panel and undocked a panel, then often they somehow close it and ask where their panel has gone or why they screen looks the way it does.

Adjusting the panel by redocking items is more of an intermediate user's action, so I think it should be protected for beginners. Therefore, my suggestion is to have the "Lock Room Panes" and rooms padlock both locked by default on first installation.

**Comment by RodneyBaker** [comment]
Agreed.   We'll likely field question of 'why can't I change this/that' but that will be better than having the chaos of broken/lost windows.

One thing we should do in the future is take a hard look at the installer (at least the Windows installer) because there may be a way to query the user for customized preferences during installation.  Then at least the user will know what is being set should they opt to know and or change the defaults during the install.

Added:  A customized/improved installer might also be a teaching/learning opportunity as it can clue a new user into what they will encounter later on and when in doubt and not sure to rerun the installer and choose different options.

Added:  Opentoonz does have one additional consideration that Tahoma does not in this regard in that Opentoonz has two places to lock windows while Tahoma has only one.  While I'm easily convinced, I'm not sure I would lock both of these by default if they aren't similarly connected.

**Comment by RodneyBaker** [comment]
For those interested in what this 'rooms and panels locked' start up workflow is like it is the default in the Tahoma2D fork of Opentoonz although there are other changes that combine rooms/panels too so that both can be locked down with one selection.

As a proposal the primary thing to consider now with regard to Opentoonz is what, if any, potential negative effects it might have on setup and workflow.

This proposal does appear to be a good candidate as a v1.5 default.  
Especially in cases where new users may be apt to change their layouts inadvertently and not know how to set them back.

**Comment by RodneyBaker** [comment]
I'm tagging this with 'roadmap' as we should be able to make a decision on whether or not to implement this in the 2021 development cycle.  Since the code is (mostly) present in Tahoma it can be a fairly direct port although some differences may require additional work.

Aside:  I personally am in favor of the proposal.
For those desiring to see what success might look like please test the feature in Tahoma2D and provide your feedback on how well it works for you.

**Comment by RodneyBaker** [comment]
@DarrenTAnims It appears we allowed this proposal to slip off the radar.
Two things should help us examine how best to proceed:  Finding specifica locations in the code that will allow the windows/panels to be locked down and examing how Tahoma2D has implemented this already (Pros and Cons).

As a house cleaning measure will close this temporarily with goal to take up the proposal again in 2022 development cycle.

**Comment by artisteacher** [comment]
On the pros end of things - I used to have students accidentally pull panels out of rooms all the time. It almost never happens with T2D since you have to deliberately unlock the workspace.

**Comment by RodneyBaker** [comment]
This relates in a way as it is important what we lock down...

There has been some discussion of late about what Opentoonz looks like when it is launched from a clean install.
Here is what I see with a clean install:

![image](https://user-images.githubusercontent.com/2576432/155863329-41aedf90-bcae-4991-89c9-2ac7d4137a61.png)

I'm curious why some have reported seeing Opentoonz differently.

This of course doesn't change the fact that intially the UI should be locked down.
It's trivial thereafter for a user to unlock it and customize.


While we wait for this to be implemented I'll suggest the workaround as:  Immediately after installing lock the UI down.

Here's the other request related to percentage positioning of panels: #4245

**Comment by gab3d** [comment]
This is how it starts fresh in a 1280x720 notebook:

![imagen](https://user-images.githubusercontent.com/18104091/155863849-12014897-db26-4f6e-9d58-6343eddb07e9.png)

**Comment by flurick** [comment]
The [Tahoma implementation](https://github.com/tahoma2d/tahoma2d/commit/67c02220a5b47afd927d01810a57f11ecc88834a) for reference.

**Comment by flurick** [comment]
Could it be [this simple](https://github.com/opentoonz/opentoonz/commit/efecb815e30b19f776b02844e909225c929f62e9)? But the tahoma lock for both rooms and docks is nice, only down side is it might take a second to realize the locking is not in the same menu for new users (?)

**Comment by RodneyBaker** [comment]
Yes, changing that 0 to a 1 should do the job.

`TEnv::IntVar LockRoomTabToggle("LockRoomTabToggle", 1);`

REF:  https://github.com/opentoonz/opentoonz/blob/aef648995374480db9ccff851f11c082fccb0d59/toonz/sources/toonz/menubar.cpp#L54

The lines directly above that are suspect:

```
UrlOpener dvHome(QUrl("http://www.toonz.com/"));
UrlOpener manual(QUrl("file:///C:/gmt/butta/M&C in EU.pdf"));
```

The other line of interest:

https://github.com/opentoonz/opentoonz/blob/aef648995374480db9ccff851f11c082fccb0d59/toonz/sources/toonz/menubar.cpp#L66
